spec/models/course/assessment_spec.rb

```diff
@@ -123,10 +126,20 @@
             answer.save
           end
 
-          it 'returns the first unanswered question' do
-            expect(assessment.questions.step(submission, 2)).
+          it 'returns the first unanswered question when index is unaccessible' do
```

Comment:
Sure, actually I have considered using context block, but thought they are a bit "overkill" for these cases.